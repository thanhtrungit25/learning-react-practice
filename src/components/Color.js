import { PropTypes, Component } from 'react'
import StarRating from './StarRating'
import '../../stylesheets/Color.scss'

class Color extends Component {

    render() {
        const {title, color, rating, onRate, onRemove} = this.props
        return (
            <section className="color">
                <h1>{title}</h1>
                <button onClick={onRemove}>X</button>
                <div className="color"
                    style={{ backgroundColor: color }}>
                </div>
                <div>
                    <StarRating starsSelected={rating} onRate={onRate} />
                </div>
            </section>
        )
    }

}


Color.propTypes = {
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    rating: PropTypes.number,
    onRemove: PropTypes.func,
    onRate: PropTypes.func,
}

export default Color