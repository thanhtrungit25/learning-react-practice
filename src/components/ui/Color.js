import { PropTypes, Component } from 'react'
import StarRating from './StarRating'
import '../../../stylesheets/Color.scss'
import TimeAgo from './TimeAgo'
import FaTrash from 'react-icons/lib/fa/trash-o'
import { removeColor, rateColor } from '../../actions'

class Color extends Component {

    render() {
        const {id, title, color, rating, timestamp, onRate, onRemove } = this.props
        const { store } = this.context
        return (
            <section className="color" style={this.style}>
                <h1 ref="title">{title}</h1>
                <button onClick={onRemove}>
                    <FaTrash />
                </button>
                <div className="color"
                    style={{ backgroundColor: color }}>
                </div>
                <TimeAgo timestamp={timestamp} />
                <div>
                    <StarRating starsSelected={rating}
                                onRate={onRate} />
                </div>
            </section>
        )
    }
}

Color.contextTypes = {
    store: PropTypes.object
}

Color.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    rating: PropTypes.number,
    onRate: PropTypes.func,
    onRemove: PropTypes.func
}

Color.defaultProps = {
    rating: 0
}

export default Color