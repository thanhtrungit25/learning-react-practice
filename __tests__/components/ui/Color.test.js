import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { compose } from 'redux'

import Color from '../../../src/components/ui/Color'

describe("<Color /> UI Component", () => {

    const shallowExpect = compose(expect, toJSON, shallow)

    it("Renders correct properties", () => {
        shallowExpect(<Color title="Test Color"
            color="Test super color"
            rating={3}
            timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
        />).toMatchSnapshot()
    })
    

})