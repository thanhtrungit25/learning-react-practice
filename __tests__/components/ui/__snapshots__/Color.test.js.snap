// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Renders correct properties 1`] = `
<section
  className="color"
>
  <h1>
    Test Color
  </h1>
  <button>
    <FaTrashO />
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "Test super color",
      }
    }
  />
  <TimeAgo
    timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
  />
  <div>
    <StarRating
      starsSelected={3}
    />
  </div>
</section>
`;
